wicket-examples/src/main/java/org/apache/wicket/examples/csp/CspApplication.java

```diff
@@ -69,4 +88,19 @@ public static String getNonce()
 		}
 		return nonce;
 	}
+
+	public static boolean isCspApplicable()
+	{
+		Request request = RequestCycle.get().getRequest();
+		if (request instanceof ServletWebRequest)
+		{
+			// Unfortunately Edge does things worse than just "doesn't support" it does support the CSP,
+			// but the 'nonce' and 'strict-dynamic' instructions were broken for ages.
+			// Edge issue https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/13246371/
+			// It's OK in new Edge chromium beta, also the new Edge has Edg/ in User-Agent header instead of Edge/
+			return !((ServletWebRequest)request).getContainerRequest().getHeader("User-Agent").contains("Edge/");
```

Comment:
Does that mean that each and every application out there needs to add this snippet of code ?
One of the reasons why CSP has been postponed for few years is that the browser support was problematic for long time.

Reply:
As of today - yes. But once new version of Edge is released 99.9 of buggy EdgeHTML/Chakra Edges supposed to be wiped out with windows update. It must happen pretty soon.

Otherwise works well in IE11, Chrome, Safari, Firefox and other modern browsers.